src/index.js

```diff
@@ -51,6 +52,14 @@ export default function babel ( options ) {
 			if ( !filter( id ) ) return null;
 			if ( id === HELPERS ) return null;
 
+			const cacheKey = `${id}|${code.length}`;
+			const digest = sha256().update(code).digest('hex');
+
+			const cache = this.cache && this.cache.get(cacheKey);
+			if (cache && digest === cache.digest) {
+				return cache.result;
+			}
```

Comment:
It would be nice to handle cache eviction by default as well to avoid eventual memory issues:

```js
else {
  this.cache.delete(cacheKey);
}
```

Reply:
This is handled implicitly once the transform is run (the transform will set the cache). Of course there is an edge case where if the transform errors then the cache is not evicted. I'm happy to add this code if you think that is enough of a concern?

Reply:
We just need to make sure we work out the cases where the cache key changes, and we orphan an entire cache and never clear it...